### ISSUE TITLE
Add more robust validation, similar to `FileField`, for `VaFileInputMultiple`

### ISSUE THREAD

**Issue by JoshingYou1** [department-of-veterans-affairs/vets-design-system-documentation#3349]
# Bug Report

- [X] I’ve searched for any related issues and avoided creating a duplicate issue.

## Reproducing

- Formation version: 1.0.1
- Device: Macbook
- Browser: Google Chrome

Steps to reproduce:

1. Create a `VaFileInputMultiple` component.
2. Add the following code to the component's attributes: `accept=".pdf, .doc, .docx, .jpg, .jpeg, .rtf, .png"`
3. Change the file extension of a `.gif` file to `.doc` and upload it. It should let allow you to upload it with no validation errors, which is not the desired outcome.

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [X] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details
`FileField` Behavior
![Image](https://github.com/user-attachments/assets/859c1f31-c523-47b2-950a-946acf737ff3)
`VaFileInputMultiple` Behavior
![Image](https://github.com/user-attachments/assets/27a8a7a8-99a5-4316-9ed8-93675845481f)

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/3349?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @powellkerry @rsmithadhoc

**Comment by jamigibbs** [comment]
Hi @JoshingYou1. What is the `FileField` component? This is not something that is currently in the VA design system. Can you link to it?

**Comment by JoshingYou1** [comment]
> Hi [@JoshingYou1](https://github.com/JoshingYou1). What is the `FileField` component? This is not something that is currently in the VA design system. Can you link to it?

Maybe component is the wrong word? Here's the link: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/fields/FileField.jsx

**Comment by micahchiang** [comment]
@JoshingYou1 - Teams do have the ability to pass custom validation functions to `VaFileInputMultiple` through the use of the `onVaMultipleChange` prop. [Here are the docs](https://design.va.gov/storybook/?path=/docs/uswds-va-file-input-multiple--docs#custom-validation) which show an example of this.

**Comment by JoshingYou1** [comment]
> @JoshingYou1 - Teams do have the ability to pass custom validation functions to `VaFileInputMultiple` through the use of the `onVaMultipleChange` prop. [Here are the docs](https://design.va.gov/storybook/?path=/docs/uswds-va-file-input-multiple--docs#custom-validation) which show an example of this. 

Yes, I've read through the docs. Is there a straight-forward way to integrate what is in `FileField` into the `onVaMultipleChange` prop though?

**Comment by micahchiang** [comment]
The prop accepts a callback and exposes an event that contains file information. What you could probably do is either use [these file validators](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/utilities/file/index.js) directly in whatever call back you pass in, or adapt them to your specific needs and then pass them in. 

FileField is pretty tightly coupled to react-json-schema, so integrating what's in that file directly probably won't work. The validators I linked to are more agnostic. 

If you're looking to use multiple file input in a form that uses react-json-schema, the forms team does provide a [va file input field](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/web-component-fields/VaFileInputField.jsx) with the necessary bindings, though I'm not sure if adapting va file input multiple is on their roadmap.

**Comment by ndsprinkle** [comment]
Following up on this, the forms team is working on a replacement for `FileField` that will be build on `VaFileInput` and `VaFileInputMultiple` and will be modeled after our other patterns that consume web components. This will have similar functionality to `FileField`, like the validation behavior explained in this ticket, and can be used in forms to accept file input(s). Closing this issue as this functionality will not be added to the web component.
